doc/routes.md

```diff
@@ -0,0 +1,168 @@
+
+- `message/:messageID`
+  : display a given message
+
+- `conversation/:conversationID/:messageID/`
+  : display conversation view, eventually scoped on a particular message
+
+- `compose`
+  : new composition message
+
+- `compose/edit/:messageID`
+  : edit message (as a draft) in composition
+
+- `edit/:messageID`
+  : same as `compose/edit/:messageID`
+
+- `reply-all/:messageID`
+  : compose message in _reply all_ mode
+
+- `reply/:messageID`
+  : compose message in _reply mode_
+
+- `forward/:messageID`
+  : compose message in forward mode
+
+### Redirects
+
+There's severals _redirects_ inside the application to switch to another view, generally after a successful action (which isn't what we want, as long as _requests_ should handle the success state - like creating a new account - which is considered as a new _action_ in flux pattern).
+
+- `checkAccount`
+  : when a new account is created, redirect to the main account message list view
+
+- `Compose#finalRedirect`
+  : when finish a _compose_, redirect the message if in a _reply_ context, or to the account message list if not
+
+- `Compose#sendActionMessage`
+  : if the message isn't sent after _compose_, redirect back to the compose edit view
+
+- `MessageListItem`
+  : redirect the whole view to a built URL in context when clicking on a message in the list
+
+- `Message#onDelete`, `ToolbarMessage#onDelete`
+  : redirect to the next conversation (or close the panel if there's not) after deletion
+
+- `Panel#renderList`
+  : redirect to the first available account if the requested account isn't available
+
+- `SearchBar#onSearchTriggered`, `SearchBar#onAccountChanged`
+  : redirect to the search view if account exists, or fallback to the first available account in list mode if there's not
+
+- `ToolbarConversation#goto*`
+  : redirect to the next / prev conversation into the list or close the panel if the target conversation isn't available
+
+- `ToolbarMessagesList#onFilterChange`
+  : redirect to the filter view according to the parameters when filters params changed
+
+
+## V2 Refactoring
+
+The v2 refactoring introduces a new router pattern[^1] which set new routes, and forbid redirects inside components.
+
+### Routes
+
+- `account/new`
+  : creates a new account
+
+- `:accountID/settings/:section`
+  : access the settings for the given account, scoped to the specified section (e.g. `gmail/settings/general`)
+
+- `:accountID/:mailbox`
+  : display the content of a mailbox in an account (i.e. the conversations list). Can be customized using _queryString_ parameters:
+  - `?filters=[:filters]`
+    : filter conversations list, can be a list of filters (e.g. `gmail/inbox?filters=starred,unread` display gmail's _inbox_ messages which are both _unread_ and _starred_)
+  - `?sort=:column:[ASC|DESC]`
+    : set the sorting order and on which criteria for the list (e.g. `gmail/inbox?sort=sender:ASC` sort messages from gmail's inbox by sender address, ascending)
+
+- `:accountID/search`
```

Comment:
We lack a route for search in all accounts, might as well make it /search and have the account has another ?filter